test/Microsoft.AspNetCore.Mvc.FunctionalTests/Infrastructure/MvcSampleFixture.cs

```diff
@@ -1,11 +1,14 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.IO;
+using System.Reflection;
+
 namespace Microsoft.AspNetCore.Mvc.FunctionalTests
 {
     public class MvcSampleFixture<TStartup> : MvcTestFixture<TStartup>
         where TStartup : class
     {
-        public MvcSampleFixture() : base("samples") { }
+        public MvcSampleFixture() : base(Path.Combine("samples", typeof(TStartup).GetTypeInfo().Assembly.GetName().Name)) { }
```

Comment:
You probably don't need the `GetTypeInfo()` call anymore. `Type.Assembly` is a thing now.